core/templates/dev/head/services/explorationContextService.js

```diff
@@ -100,6 +100,21 @@ oppia.factory('explorationContextService', [
       isInExplorationEditorMode: function() {
         return (this.getPageContext() === PAGE_CONTEXT.EDITOR &&
             this.getEditorTabContext() === EDITOR_TAB_CONTEXT.EDITOR);
+      },
+
+      // Following functions help in remembering whether video is already
```

Comment:
For the new service, I suggest the following documentation:

```
  // In the exploration player, a video should only autoplay when it is first seen on a
  // new card, and not when the learner clicks back to previous cards in their exploration
  // playthrough. This service maintains a list of videos that have been played, so that we know
  // not to autoplay them on a second pass.
  //
  // Caveat: if the same video is shown twice in the exploration, the second and subsequent
  // instances of that video will not autoplay. We believe this occurrence is rare, and have not
  // accounted it for it here. If it turns out to be an issue, we may need to instead assign a
  // unique id to each rich-text component and use that id instead to determine whether
  // to suppress autoplaying.
```